week2/exercise P1008 in LuoGu.cpp

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+
+int isvalid(int Double, int Triple, int Flag);
+
+int main()
+{
+	int Num, Double, Triple, Flag=0;
+	for (int i = 1; i < 10; i++) {
+		for (int j = 1; j < 10; j++) {
+			if (i == j) {
+				continue;
+			} else {
+				for (int k = 1; k < 10; k++) {
+					if (i == k || j == k) {
+						continue;
+					} else {
+						Num = i * 100 + j * 10 + k; // Get a three-digit number that is different for each digit.
+						Double = Num * 2;
+						Triple = Num * 3;             // Get three numbers in the ratio of 1:2:3
+						Flag=isvalid(Double, Triple, Flag);
+						if (Flag == 0) {
+							continue;
+						} else {
+							// Output everything in ascending order.
+							printf("%d %d %d\n", Num, Double, Triple);
+						}
+					}
+				}
+			}
+		}
+	}
+	return 0;
+ } 
+ 
+ int isvalid(int Double, int Triple, int Flag)

```

Comment:
This is a very bad design. First of all, `Flag` is not used at all thus it should be removed. More importantly, this function is hard to be re-used.

A good function should be:
1. short
2. modulized. This means, the function can be easily to be used in other program with less or even no modification. This also means, it should focus on its own task and should be independent of the main task of current program.

Let's say, now we have a new requirement: given several numbers (maybe 100 or even 1000), for each number, determine whether it is a valid hundreds digit and it satisfies that the numbers in ones, tens and hundreds places are all different. Can you extend this function to easily handle such scenario?

A good design can be:
```c
int is_valid_hundreds_digit(int num) {
    if (num >= 100
        && num < 1000
        && num % 10 != (num / 10) % 10
        && num % 10 != num / 100
        && (num / 10) % 10 != num / 100) {
        return 1;
    }
    
    return 0;
}
```

or even better:
```c
int is_valid_hundreds_digit(int num) {
    if (is_hundreds(num) && has_different_digits(num)) {
        return 1;
    }
    
    return 0;
}

int is_hundreds(int num) {
    if (num >= 100 && num < 1000) {
        return 1;
    }

    return 0;
}

int has_different_digits(int num) {
    if (num % 10 != (num / 10) % 10
        && num % 10 != num / 100
        && (num / 10) % 10 != num / 100) {
        return 1;
    }

    return 0;
}
```

Then the caller can be `is_valid_hundreds_digit(double_) && is_valid_hundreds_digit(triple)`